pkg/kv/kvserver/client_lease_test.go

```diff
@@ -1035,10 +1035,16 @@ func TestLeasePreferencesDuringOutage(t *testing.T) {
 		}
 		return nil
 	})
-	_, _, enqueueError := tc.GetFirstStoreFromServer(t, 0).
-		Enqueue(ctx, "replicate", repl, true /* skipShouldQueue */, false /* async */)
 
-	require.NoError(t, enqueueError, "failed to enqueue replica for replication")
+	testutils.SucceedsSoon(t, func() error {
+		// n1 requires the lease to process the replica through the replicate
+		// queue. It will try and acquire the lease if it doesn't have it. It is
+		// possible that another node gets there before n1 - expect the acquisition
+		// to succeed eventually.
+		_, _, enqueueError := tc.GetFirstStoreFromServer(t, 0).
```

Comment:
Will this ever succeed? If we lost lease to some other store in general case we try redirectOnOrAcquireLease() which will never steal the lease if it is valid. It is not bypassed by skipShouldQueue.

If that is correct, we should enqueue on the store where the lease is if we got a lease error.

Reply:
In the failures I've reproduced, the lease is invalid. The node which beats n1 to the lease is artificially dead. 

```
replica [n1,s1,r53/1:/{Table/100-Max}] does not have the range lease: [NotLeaseHolderError] lease acquisition attempt lost to another lease, which has expired in the meantime; r53: replica (n1,s1):1 not lease holder; lease holder unknown
```


I think you're right, though. If it happened to move to one of the live replicas in the range (n4) then we'd need to enqueue there instead.

With this patch I've stressed:

```
11579 runs so far, 0 failures, over 52m30s
```

vs without

```
3928 runs completed, 1 failures, over 17m33s
```

I'll continue to let the patch stress overnight. I